flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/catalog/hive/factories/HiveCatalogFactoryTest.java

```diff
@@ -54,9 +65,49 @@ public void test() {
 		checkEquals(expectedCatalog, (HiveCatalog) actualCatalog);
 	}
 
+	@Test
+	public void testLoadHDFSConfigFromEnv() throws IOException {
+		final String k1 = "what is connector?";
+		final String v1 = "Hive";
+		final String catalogName = "HiveCatalog";
+
+		// set HADOOP_CONF_DIR env
+		final File hadoopConfDir = tempFolder.newFolder();
+		final File hdfsSiteFile = new File(hadoopConfDir, "hdfs-site.xml");
+		writeProperty(hdfsSiteFile, k1, v1);
+		final Map<String, String> originalEnv = System.getenv();
+		final Map<String, String> newEnv = new HashMap<>(originalEnv);
+		newEnv.put("HADOOP_CONF_DIR", hadoopConfDir.getAbsolutePath());
+		CommonTestUtils.setEnv(newEnv);
+
+		// create HiveCatalog use the Hadoop Configuration
+		final CatalogDescriptor catalogDescriptor = new HiveCatalogDescriptor();
+		final Map<String, String> properties = catalogDescriptor.toProperties();
+		final HiveCatalog hiveCatalog = (HiveCatalog) TableFactoryService.find(CatalogFactory.class, properties)
+			.createCatalog(catalogName, properties);
+		final HiveConf hiveConf = hiveCatalog.getHiveConf();
+		// set the Env back
+		CommonTestUtils.setEnv(originalEnv);
```

Comment:
I'd suggest move this to a finally block, so that if this test case fails, it doesn't pollute env for other tests.

Reply:
good catch !

Reply:
@lirui-apache update the test case. please take a look at it